main.go

```diff
@@ -3,24 +3,39 @@ package main
 import (
 	"crypto/tls"
 	"fmt"
-	"io"
-	"log"
 	"net"
 	"net/smtp"
+	"net/textproto"
 	"os"
+	"os/signal"
 	"regexp"
 	"strings"
+	"syscall"
 	"time"
 
 	"github.com/chrj/smtpd"
+	"github.com/google/uuid"
 )
 
+func observeErr(err smtpd.Error) smtpd.Error {
+	errorsCounter.WithLabelValues(fmt.Sprintf("%v", err.Code)).Inc()
+
+	return err
+}
+
 func connectionChecker(peer smtpd.Peer) error {
+	if *allowedSender == "" {
```

Comment:
Shouldn't this have been `*allowedNets`?

This will be fixed in upstream decke/smtprelay#18.

(Cross-posted [on the original PR](https://github.com/grafana/smtprelay/pull/4/files#r593717901)).